feat(regular_expression)!: Support ES2025 Duplicated named capture groups

leaysgur opened the pull request:
Closes #6358 

@preyneyv I know you've been working on this problem.

This is an implementation that has been dormant on my local for a while.

- All tests are passing
- However, the approach is simple but not general, so there might be some edge cases that were missed
- There's also room for improvement in terms of performance

For these reasons, it was marked as WIP for me.

I believe the test cases and other parts are usable, so feel free to fork and replace them with your implementation if you'd like.

preyneyv commented:
Awesome, thank you!

Boshen commented on crates/oxc_regular_expression/README.md:
```diff
@@ -5,3 +5,8 @@ Implements ECMAScript® 2024 Language Specification
 - https://tc39.es/ecma262/2024/multipage/text-processing.html#sec-regexp-regular-expression-objects
 - https://tc39.es/ecma262/2024/multipage/additional-ecmascript-features-for-web-browsers.html#sec-regular-expressions-patterns
+
+And, ES2025 proposals
```
```suggestion
And, Stage 3 proposals
```

Boshen reviewed (commented):
NICE!

Boshen commented:
Is this in a mergable state?

leaysgur commented:
Yes, mergable. 

@preyneyv feel free to submit your work as an enhancement after merging. 😉

preyneyv commented:
Sounds good! I _think_ I solved the general problem _on paper_, should hopefully have smth up in a bit

Boshen commented:
### Merge activity

* **Oct 24, 10:13 PM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

leaysgur committed:
feat(regular_expression)!: Support ES2025 Duplicated named capture groups (#6847)

Closes #6358

@preyneyv I know you've been working on this problem.

This is an implementation that has been dormant on my local for a while.

- All tests are passing
- However, the approach is simple but not general, so there might be some edge cases that were missed
- There's also room for improvement in terms of performance

For these reasons, it was marked as WIP for me.

I believe the test cases and other parts are usable, so feel free to fork and replace them with your implementation if you'd like.
